pmoves/tools/monitoring_smoke.py

```
@@ -213,16 +250,27 @@ def main() -> int:
+        fallback_names = _provisioned_datasource_names(repo_root)
+        if fallback_names:
+            datasource_names = fallback_names
+            state.warn(f"Grafana datasource API unavailable; used provisioning fallback ({exc})")
     has_prom = "prometheus" in datasource_names
     has_loki = "loki" in datasource_names
 
     try:
         dash_payload = _http_json(f"{grafana}/api/search?type=dash-db", auth=(guser, gpass))
         dash_count = len(dash_payload) if isinstance(dash_payload, list) else 0
-    except Exception:
+    except Exception as exc:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)